src/eddymotion/model/base.py

```diff
@@ -177,16 +344,21 @@ def fit(self, data, n_jobs=None, **kwargs):
         self._is_fitted = True
         self._model = None  # Preempt further actions on the model
 
-    def predict(self, gradient, **kwargs):
+    def predict(self, gradient=None, **kwargs):
         """Predict asynchronously chunk-by-chunk the diffusion signal."""
 
+        if gradient is None:
+            raise ValueError("A gradient to be simulated (b-vector, b-value) must be provided")
+
         if not self._is_fitted:
             raise ModelNotFittedError(f"{type(self).__name__} must be fitted before predicting")
 
-        if self._b_max is not None:
-            gradient[-1] = min(gradient[-1], self._b_max)
+        gradient = np.array(gradient)  # Tuples are unmutable
+
+        # Cap the b-value if b_max is defined
+        gradient[-1] = min(gradient[-1], self._b_max or gradient[-1])
 
-        gradient = _rasb2dipy(gradient)
+        self._gtab = _rasb2dipy(gradient)
```

Comment:
We should not override the gradient table in predict:

```suggestion
        gradient = _rasb2dipy(gradient)
```